### PR TITLE
Auth first steps

### PR DESCRIPTION
**Pull request by grenlayk**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+@WebServlet("/home")
+public class HomeServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json;");
+
+    UserService userService = UserServiceFactory.getUserService();
+    JSONObject user = new JSONObject();
```
Comment (zajonc-gugle) [inline portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java:34]: One thing I can suggest is that it's usually a good idea to be explicit in naming, especially when it comes to describing structures that are going to be sent across different modules of the app (here, between the backend service and client script).

The object returned from this method can represent two cases:
- the user is logged in
- the user is not logged in

In both cases the object we return has the same shape: `{ email, url }`. That doesn't have to be the case, and you're free to return different objects for different cases. Actually, in many cases that's kind of desired - you'd rather have your frontend app crash on referencing an invalid field, than have it use a field thinking that it means something else than it does (e.g. using a `logoutUrl` in place of a `loginUrl`). In your case the difference is not harmful, but you can be more explicit and put the logoutUrl under a `logoutUrl` prop, and loginUrl under `loginUrl`.

File: portfolio/src/main/java/com/google/sps/servlets/DeleteMessagesServlet.java [hunk 2]
```diff
@@ -32,22 +32,36 @@
...
-@WebServlet("/delete-messages")
+@WebServlet("/delete-message")
 public class DeleteMessagesServlet extends HttpServlet {
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("userMessage").addSort("timestamp", SortDirection.DESCENDING);
+
+    Long id = Long.parseLong(request.getParameter("id"));
+    String messageUserEmail = request.getParameter("userEmail");
```
Comment (bprusak) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteMessagesServlet.java:46]: This is a potential security breach.
You need to assume someone will be willing to bypass the frontend and type the request themselves. They may go ahead and pass an `id` of a message they don't own along with `messageUserEmail` equal to email they do own.
The best solution would be to accept only the `id` of the message, then query the datastore to get this message author's email.

File: portfolio/src/main/java/com/google/sps/servlets/DeleteMessagesServlet.java [hunk 3]
```diff
@@ -32,22 +32,36 @@
...
+
+    Key messageEntityKey = KeyFactory.createKey("userMessage", id);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);
+    UserService userService = UserServiceFactory.getUserService();
+
+    if (!userService.isUserLoggedIn())  {
+        response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+        response.getWriter().println("You should log in to delete messages");
+        //response.sendRedirect("/user-info");
```
Comment (zajonc-gugle) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteMessagesServlet.java:55]: Let's try to remove commented out code.

File: portfolio/src/main/java/com/google/sps/servlets/DeleteMessagesServlet.java [hunk 4]
```diff
@@ -32,22 +32,36 @@
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
 
 
-@WebServlet("/delete-messages")
+@WebServlet("/delete-message")
 public class DeleteMessagesServlet extends HttpServlet {
```
Comment (zajonc-gugle) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteMessagesServlet.java:40]: It's usually a good idea to be consistent.
Let's keep the class name, file name and servlet path the same. If the servlet deletes a single message, let's rename the class to `DeleteMessageServlet` (singular)

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -19,21 +19,52 @@ async function loadMessages() {
...
+function createMessageElement(message, userEmail) {
+    const messageElement = document.createElement('li');
     const curMessage = document.createElement("p");
-    curMessage.appendChild(createMyElement(message.userName + ": ", "b"));
+    curMessage.appendChild(createMyElement(message.userEmail + ": ", "b"));
     curMessage.appendChild(createMyElement(message.userMessage, "bdi"));
 
-    messagesEl.appendChild(curMessage);
-  }
+    console.log("mes: " + message.userEmail);
```
Comment (zajonc-gugle) [inline portfolio/src/main/webapp/script.js:39]: Let's get rid of debug logs.

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -19,21 +19,52 @@ async function loadMessages() {
...
+    console.log("cur:" + userEmail);
+
+    if (message.userEmail == userEmail) {
+        const deleteButtonElement = document.createElement('button');
+        deleteButtonElement.innerText = 'Delete';
+        deleteButtonElement.addEventListener('click', () => {
+        deleteMessage(message);
```
Comment (zajonc-gugle) [inline portfolio/src/main/webapp/script.js:46]: Indentation is a bit off in those 2 lines.

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -141,3 +172,79 @@ function addLandmark(map, lat, lng, title, description) {
...
+    userEl.appendChild(createMyElement("Hi, stranger!   ", 'bdi'));
+    const aEl = createMyElement("Log in", 'a');
+    aEl.setAttribute('href', loginUrl);
+    buttonEl.appendChild(aEl);
+  }
+  userEl.appendChild(buttonEl);
+}
+
+function createForm() {
+
```
Comment (zajonc-gugle) [inline portfolio/src/main/webapp/script.js:209]: Usually there's no blank line after function name.

**Pull request merged**